src/openrct2/config/Config.cpp

```diff
@@ -344,8 +344,20 @@ namespace Config
     {
         if (reader->ReadSection("network"))
         {
+            // If the `player_name` setting is missing or equal to the empty string
+            // use the logged-in user's username instead
+            utf8* playerName = reader->GetCString("player_name", "");
+            if (String::Compare(playerName, "") == 0)
```

Comment:
Ahhh, I thought there was something off about my condition. That'll be it. (Can you tell I'm better at C than C++? 😅)